app/controllers/amends/ViewReturnSummaryController.scala

```diff
@@ -57,11 +57,11 @@ class ViewReturnSummaryController @Inject() (
           submittedReturn <- submittedReturnF
           obligation <- fulfilledObligationF
           updatedAnswers <- Future.fromTry(
-            request.userAnswers.getOrElse(UserAnswers(request.userId)).set(
+            request.userAnswers.getOrElse(UserAnswers(request.cacheKey)).set(
```

Comment:
Can we make userAnswers not optional on the optionalRequest, and do the getOrElse in there, so that the creation of this blank user answers only happens there, and cant be done with anything but the cacheKey

Reply:
But this is an optional data request? By its very name it means the data may not be present? If we do that we may swell use the DataRequest?

Reply:
Ok. Its the same really. Not got my thinking cap on. Making the change :-)